org.eclipse.ui.tests.decorators.DecoratorAdaptableTest.testAdaptables is failing on linux

In the Build [I20230405-1800](https://download.eclipse.org/eclipse/downloads/drops4/I20230405-1800)  the below tests in org.eclipse.ui.tests.decorators.DecoratorAdaptableTests failed

on [cen64](https://download.eclipse.org/eclipse/downloads/drops4/I20230405-1800/testresults/html/org.eclipse.ui.tests_ep428I-unit-cen64-gtk3-java17_linux.gtk.x86_64_17.html)
```
Adaptable test 1 has failed for object org.eclipse.ui.tests.menus.ObjectContributionClasses$A@728a2652
java.lang.AssertionError: Adaptable test 1 has failed for object org.eclipse.ui.tests.menus.ObjectContributionClasses$A@728a2652
at org.junit.Assert.fail(Assert.java:89)
at org.junit.Assert.assertTrue(Assert.java:42)
at org.eclipse.ui.tests.decorators.DecoratorAdaptableTests.assertDecorated(DecoratorAdaptableTests.java:56)
at org.eclipse.ui.tests.decorators.DecoratorAdaptableTests.testAdaptables(DecoratorAdaptableTests.java:84)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
```

--- a/tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/decorators/DecoratorAdaptableTests.java
+++ b/tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/decorators/DecoratorAdaptableTests.java
@@ -25,6 +25,7 @@ import org.eclipse.ui.internal.WorkbenchPlugin;
 import org.eclipse.ui.internal.decorators.DecorationResult;
 import org.eclipse.ui.internal.decorators.DecoratorManager;
 import org.eclipse.ui.internal.decorators.LightweightDecoratorManager;
+import org.eclipse.ui.tests.harness.util.UITestUtil;
 import org.eclipse.ui.tests.menus.ObjectContributionClasses;
 import org.junit.After;
 import org.junit.Before;
@@ -63,6 +64,10 @@ public class DecoratorAdaptableTests {
 		PlatformUI.getWorkbench().getDecoratorManager().setEnabled(TestUnadaptableDecoratorContributor.ID, true);
 		PlatformUI.getWorkbench().getDecoratorManager().setEnabled(TestResourceDecoratorContributor.ID, true);
 		PlatformUI.getWorkbench().getDecoratorManager().setEnabled(TestResourceMappingDecoratorContributor.ID, true);
+
+		// Process all pending UI events to ensure decorator enablement has completed
+		// This prevents race conditions where decorators may not be fully registered yet
+		UITestUtil.processEvents();
 	}
 
 	@After
@@ -71,6 +76,9 @@ public class DecoratorAdaptableTests {
 		PlatformUI.getWorkbench().getDecoratorManager().setEnabled(TestUnadaptableDecoratorContributor.ID, false);
 		PlatformUI.getWorkbench().getDecoratorManager().setEnabled(TestResourceDecoratorContributor.ID, false);
 		PlatformUI.getWorkbench().getDecoratorManager().setEnabled(TestResourceMappingDecoratorContributor.ID, false);
+
+		// Process all pending UI events to ensure clean state for next test
+		UITestUtil.processEvents();
 	}
 
 	/**

